stellarphot/photometry/source_detection.py

```diff
@@ -111,13 +70,31 @@ def compute_fwhm(
         The FWHM of each source in the x and y directions.
 
     """
+    if sky_per_pix_avg is not None and sky_per_pix_column is not None:
+        raise ValueError(
+            "Cannot specify both `sky_per_pix_avg` and `sky_per_pix_column`."
+        )
+
+    if sky_per_pix_column is not None:
+        if sky_per_pix_column not in sources.colnames:
+            raise ValueError(f"Column {sky_per_pix_column} not found in sources table.")
+        sky_values = sources[sky_per_pix_column]
+
+    if sky_per_pix_avg is not None:
+        # User gave a value for the sky background to subtract
+        sky_values = [sky_per_pix_avg] * len(sources)
+    elif sky_per_pix_column is None:
+        # User didn't give a value for the sky background to subtract
+        # so try an of estimate it from the image
+
+        sky_values = [np.nanmedian(ccd.data)] * len(sources)
 
     fwhm_x = []
     fwhm_y = []
-    for source in sources:
+    for source, sky in zip(sources, sky_values, strict=True):
```

Comment:
If you have strict=True, you might want to wrap this for loop in a try/except block to handle ValueError crashes, or are you handling that when calling the function?

Reply:
Right now I don't think it is getting handled at all; I'm not sure special handling is needed since this is likely an error in the user's data, not an error in what we do with the data. 

I do need to bump the minimum version of Python since `strict` was introduced in 3.11....

Reply:
Bumped the minimum python version...